src/tox_lsr/test_scripts/runcontainer.sh

```diff
@@ -102,24 +102,48 @@ refresh_test_container() {
 
     # shellcheck disable=SC2086
     if [ "$erase_old_snapshot" = true ] || [ "$created" -lt "$age" ]; then
-        container_id=$(podman run -d $CONTAINER_OPTS ${LSR_CONTAINER_OPTS:-} \
-            $CONTAINER_MOUNTS "$CONTAINER_BASE_IMAGE" "$CONTAINER_ENTRYPOINT")
-        if [ -z "$container_id" ]; then
-            echo ERROR: Failed to start container
-            return 1
-        fi
-        # shellcheck disable=SC2064
-        trap "podman rm -f $container_id" RETURN
-        local pkgcmd prepkgs
+        local pkgcmd prepkgs initpkgs image_name
         case "$CONTAINER_IMAGE_NAME" in
         *-7) pkgcmd=yum ;
+             initpkgs="" ;
              prepkgs="" ;;
         *-8) pkgcmd=dnf ;
+             initpkgs="" ;
              prepkgs="" ;;
         *-9) pkgcmd=dnf ;
+             initpkgs=systemd ;
              prepkgs="dnf-plugins-core" ;;
         *) pkgcmd=dnf; prepkgs="" ;;
         esac
+        image_name="$CONTAINER_BASE_IMAGE"
+        if [ -n "${initpkgs:-}" ]; then
+            # some images do not have the entrypoint, so that must be installed
+            # first
+            container_id=$(podman run -d $CONTAINER_OPTS ${LSR_CONTAINER_OPTS:-} \
+                $CONTAINER_MOUNTS "$image_name" sleep 3600)
+            if [ -z "$container_id" ]; then
+                echo ERROR: Failed to start container
+                return 1
+            fi
+            if ! podman exec -i "$container_id" "$pkgcmd" install -y $initpkgs; then
+                podman rm -f "$container_id"
```

Comment:
Maybe it should be extremely rare, but do we need to have an error message like the line 125 here and next to the line 133? Or it's ok since `podman exec` or `podman container` command-line issues its error message to stderr?

Other than that, lgtm.

Reply:
The error message should go to stderr here.